packages/SwingSet/test/upgrade/vat-ulrik-1.js

```diff
@@ -167,7 +168,78 @@ export const buildRootObject = (_vatPowers, vatParameters, baggage) => {
     returnEternalPromise: () => p2,
 
     makeLostKind: () => {
-      makeKindHandle('unhandled', []);
+      makeKindHandle('unhandled');
+    },
+
+    makeMultiKind: mode => {
+      const mkh = makeKindHandle('multi');
+      baggage.init('mkh', mkh);
+      switch (mode) {
+        case 's2mFacetiousnessMismatch': {
+          // upgrade should fail
+          defineDurableKind(mkh, () => ({}), {
+            fooMethod: () => 1,
+          });
+          break;
+        }
+        case 'facetCountMismatch': {
+          // upgrade should fail
+          defineDurableKindMulti(mkh, () => ({}), {
+            foo: {
+              fooMethod: () => 1,
+            },
+            bar: {
+              barMethod: () => 1,
+            },
+          });
+          break;
+        }
+        case 'facetNameMismatch': {
+          // upgrade should fail
+          defineDurableKindMulti(mkh, () => ({}), {
+            foo: {
+              fooMethod: () => 1,
+            },
+            bar: {
+              barMethod: () => 1,
+            },
+            belch: {
+              belchMethod: () => 1,
+            },
+          });
+          break;
+        }
+        case 'facetOrderMismatch': {
+          // upgrade should succeed
```

Comment:
maybe augment the comment, `upgrade should succeed because we sort() facet names`